src/builder.ts

```diff
@@ -102,16 +121,20 @@ export default class Builder {
 
 			// Catch any errors the stream produces
 			outputStream.on('error', (err: Error) => {
+				errored = true
 				self.callHook(hooks, 'buildFailure', handler, err, self.layers)
 			})
 			dup.on('error', (err: Error) => {
+				errored = true
 				self.callHook(hooks, 'buildFailure', handler, err, self.layers)
 			})
 
 			// Setup the buildSuccess hook. This handler is not called on
 			// error so we can use it to propagate the success information
 			outputStream.on('end', () => {
-				this.callHook(hooks, 'buildSuccess', handler, _.last(this.layers), this.layers)
+				if (!errored) {
```

Comment:
I think a better approach to this would be to remove this handler when an error occurs instead